### PR TITLE
Bug 1481147 - Fix default pod image for network diagnostics

### PR DESCRIPTION
**Pull request by pravisankar**
- This also ensures network diagnostics pod and test pod images uses
  deployed openshift version/tag (not the latest) so that it doesn't
  need to download another same image with latest tag.

- Print more details when network diagnostics test setup fails.
  Currently when network diags fails, it only informs how many test pods failed but doesn't provide why those pods failed. This change will fetch logs for the pods in case of setup failure.

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/networking @knobunc PTAL

**Comment by pravisankar** [comment]
[test]

**Comment by jcantrill** [comment]
@sosiouxme Should this be assigned to you?

**Comment by sosiouxme** [comment]
looks like a flake
/test cmd

As an assignee, what am I supposed to do? Review only? Give a /lgtm? Doesn't look like I need to /approve.

**Comment by pravisankar** [comment]
@sosiouxme merge-bot says this pr needs an approver from pkg/diagnostics/OWNERS and pkg/oc/OWNERS
You are one of the owners in pkg/diagnostics, so I think you can just review or can review, give /lgtm and/or /approve.

File: pkg/diagnostics/network/objects.go [hunk 1]
```diff
@@ -131,6 +123,14 @@ func GetTestPod(testPodImage, testPodProtocol, podName, nodeName string, testPod
 	return pod
 }
 
+func getTrimmedImage(image string) string {
+	// Image format could be: [<dns-name>/]openshift/origin-deployer[:<tag>]
+	tokens := strings.Split(image, "/")
```
Comment (knobunc) [inline pkg/diagnostics/network/objects.go]: Can't you use https://golang.org/pkg/strings/#SplitN

Comment (pravisankar) [reply pkg/diagnostics/network/objects.go]: We don't know how many `/`s will be present in the given image (dns path may have many) and we only care about the last token.

Comment (knobunc) [reply pkg/diagnostics/network/objects.go]: Oh, the last.  Gotcha!

**Comment by sosiouxme** [comment]
@pravisankar you're also an approver for that path so my approval is superfluous :)

**Pull request merged**